src/AutoRest.CodeModel/Program.cs

```diff
@@ -40,40 +42,13 @@ private static void Main()
                                 && !l.Contains("Newtonsoft.Json.JsonExtensionData"))
                     .Select(l => Regex.Replace(l, @"(.*\[)Newtonsoft\.Json\.JsonProperty\((.*""),?.*(\)\])",
                         "$1YamlDotNet.Serialization.YamlMember(Alias = $2$3", RegexOptions.Singleline).TrimEnd()))
-                // Workaround for anyOf SecurityScheme types
+                // Removes AdditionalProperties property from types as they are required to derive from IDictionary<string, object> for deserialization to work properly
                 .Replace($"        private System.Collections.Generic.IDictionary<string, object> _additionalProperties = new System.Collections.Generic.Dictionary<string, object>();{Environment.NewLine}{Environment.NewLine}        public System.Collections.Generic.IDictionary<string, object> AdditionalProperties{Environment.NewLine}        {{{Environment.NewLine}            get {{ return _additionalProperties; }}{Environment.NewLine}            set {{ _additionalProperties = value; }}{Environment.NewLine}        }}{Environment.NewLine}", String.Empty)
```

Comment:
This was still needed as it adds AdditionalProperties to these types. Language does it's own implementation of this so that the deserializer doesn't fail when there are unknown keys.